cudax/test/event/event_smoke.cu

```diff
@@ -0,0 +1,129 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#include <cuda/experimental/event.cuh>
+
+#include "../common/utility.cuh"
+#include "../hierarchy/testing_common.cuh"
+#include <catch2/catch.hpp>
+
+namespace
+{
+namespace test
+{
+cudax::event_ref fn_takes_event_ref(cudax::event_ref ref)
+{
+  return ref;
+}
+} // namespace test
+} // namespace
+
+static_assert(!_CUDA_VSTD::is_default_constructible_v<cudax::event_ref>);
+static_assert(!_CUDA_VSTD::is_default_constructible_v<cudax::event>);
+static_assert(!_CUDA_VSTD::is_default_constructible_v<cudax::timed_event>);
+
+TEST_CASE("can construct an event_ref from a cudaEvent_t", "[event]")
+{
+  ::cudaEvent_t ev;
+  CUDAX_REQUIRE(::cudaEventCreate(&ev) == ::cudaSuccess);
```

Comment:
There is a CUDART macro in testing_common (I need to move that header) that check if a runtime calls was successful